app/src/include/firebase/app.h

```diff
@@ -701,6 +701,23 @@ class App {
   static void SetDefaultConfigPath(const char* path);
 #endif  // INTERNAL_EXPERIMENTAL
 
+#ifdef INTERNAL_EXPERIMENTAL
+  // These methods are only visible to SWIG and internal users of firebase::App.
+
+  /// On Desktop, log a Heartbeat
+  void LogHeartbeat();
+
+  /// On Desktop, clear stored heartbeats and return a payload.
+  ///
+  /// @return encoded payload of all logged heartbeats.
+  std::string GetAndResetStoredHeartbeats();
+
+  /// On Desktop, clear today's stored heartbeat and return a payload.
+  ///
+  /// @return encoded payload of today's logged heartbeat.
+  std::string GetAndResetTodaysStoredHeartbeats();
+#endif  // INTERNAL_EXPERIMENTAL
+
```

Comment:
Since these are used for desktop platform only, I feel it is a bit confusing that this API is available for iOS and Android as well.

To reduce the confusion, I would recommend:
1. Wrap this inside `#if FIREBASE_PLATFORM_WINDOWS | FIREBASE_PLATFORM_OSX | FIREBASE_PLATFORM_LINUX` and remove the stub impl for Android, iOS and stub.
2. Renaming the functions to be more specific, ex. `LogDesktopHeartbeat`, `GetAndResetStoredDesktopHeartbeats` and `GetAndResetTodaysStoredDesktopHeartbeats`

Reply:
Done